### PR TITLE
Implement `toDot` Method and File Storing Mechanism for `VerkleTrie`

### PR DESCRIPTION
**Pull request by neotheprogramist**
fixes https://github.com/neotheprogramist/besu-verkle-trie/issues/3

### CODE REVIEW

**Comment by Uacias** [comment]
The flag showRepeatingEdges (here I admit that the variable name does not accurately convey the actual intention) was supposed to be responsible for eliminating the edges that disrupted the visibility of the graph, for example, (InternalNode → NullNode, e.g., 240 times).

After carefully considering the feedback, I  certainly agree with the suggestion to change the flag to showNullNodes and modify the functionality to allow displaying the tree with NullNodes/NullLeafNodes or without them.

Thank you for bringing this to my attention, and a revised version in accordance with the feedback will be provided soon.

**Review by thomas-quadratic — COMMENTED** [review]
Thanks, I tested it and it seems to work fine.
I think we can merge it now (although I do not have those rights).

**Review by matkt — APPROVED** [review]

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java [hunk 1]
```diff
@@ -25,6 +25,66 @@
...
+        Bytes32.fromHexString("0x1000000000000000000000000000000000000000000000000000000000000000");
+    Bytes32 key2 =
+        Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddee00");
+    Bytes32 value2 =
+        Bytes32.fromHexString("0x0100000000000000000000000000000000000000000000000000000000000000");
+
+    trie.put(key1, value1);
+    trie.put(key2, value2);
+
+    System.out.println(trie.toDotTree(true));
```
Comment (matkt) [inline src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java:85]: could we have an assert ?

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java [hunk 2]
```diff
@@ -25,6 +25,66 @@
...
+  @Test
+  public void testToDotTrieOneValueRepeatingEdges() {
+    SimpleVerkleTrie<Bytes32, Bytes32> trie = new SimpleVerkleTrie<>();
+    Bytes32 key =
+        Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff");
+    Bytes32 value =
+        Bytes32.fromHexString("0x1000000000000000000000000000000000000000000000000000000000000000");
+    trie.put(key, value);
+
+    System.out.println(trie.toDotTree(true));
```
Comment (matkt) [inline src/test/java/org/hyperledger/besu/ethereum/trie/verkle/SimpleVerkleTrieTest.java:67]: same for all test

File: src/test/java/org/hyperledger/besu/ethereum/trie/verkle/DotDisplayTest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    Bytes32 key =
+        Bytes32.fromHexString("0x00112233445566778899aabbccddeeff00112233445566778899aabbccddeeff");
+    Bytes32 value =
+        Bytes32.fromHexString("0x1000000000000000000000000000000000000000000000000000000000000000");
+    trie.put(key, value);
+
+    final String fileName = "expectedTreeOneValueShowNulls.txt";
+    final String expectedTree = getResources(fileName);
+    final String actualTree = trie.toDotTree(true);
+    // System.out.println(actualTree);
```
Comment (matkt) [inline src/test/java/org/hyperledger/besu/ethereum/trie/verkle/DotDisplayTest.java:99]: could we remove that

**Review by matkt — CHANGES_REQUESTED** [review]
Just fixing this small comment and I will merge it. thanks for your PR

**Pull request merged**